packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -2015,6 +2021,10 @@ class _TimePickerDialogState extends State<TimePickerDialog> with RestorationMix
           _autofocusMinute.value = false;
           _entryMode.value = TimePickerEntryMode.dial;
           break;
+        case TimePickerEntryMode.dialOnly:
+        case TimePickerEntryMode.inputOnly:
+          assert(false, 'Can not change entry mode from _entryMode');
```

Comment:
Instead of an assert it would be better to use a `FlutterError` here. Take a look for other examples that use this. Also, `_entryMode` in the string needs a `$` in front of it, or the error message will literarily be: `Can not change entry mode from _entryMode'.